app/src/main/java/com/eveningoutpost/dexdrip/utils/Preferences.java

```diff
@@ -417,14 +442,18 @@ protected void onResume()
             LocationHelper.requestLocationForBluetooth(this); // double check!
         }
         PreferenceManager.getDefaultSharedPreferences(this).registerOnSharedPreferenceChangeListener(LeFunEntry.prefListener);
+        PreferenceManager.getDefaultSharedPreferences(this).registerOnSharedPreferenceChangeListener(MiBandEntry.prefListener);
         PreferenceManager.getDefaultSharedPreferences(this).registerOnSharedPreferenceChangeListener(BlueJayEntry.prefListener);
+        LocalBroadcastManager.getInstance(this).registerReceiver(mibandStatusReceiver,
```

Comment:
Does this need an associated `unregisterReceiver` in `onPause()` ?

Reply:
Yes, you right, this is a good suggestion to deregister this receiver when activity paused.

Reply:
If you can update the PR with this change I can approve it for merging. thanks

Reply:
Ok, done